src/github.com/getlantern/lantern-android/client/config.go

```diff
@@ -0,0 +1,122 @@
+package client
+
+import (
+	"compress/gzip"
+	"crypto/x509"
+	"errors"
+	"github.com/getlantern/fronted"
+	"github.com/getlantern/keyman"
+	"github.com/getlantern/yaml"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type clientCfg struct {
+	FrontedServers []frontedServer                  `yaml:"frontedservers"`
+	MasqueradeSets map[string][]*fronted.Masquerade `yaml:"masqueradesets"`
+}
+
+// Config provides client configuration.
+type Config struct {
+	Client     clientCfg `yaml:"client"`
+	TrustedCAs []*CA     `yaml:"trustedcas"`
+}
+
+var (
+	// ErrFailedConfigRequest is returned when the server replies with a non-200
+	// status code to our request for a configuration file.
+	ErrFailedConfigRequest = errors.New(`Could not get configuration file.`)
+
+	// ErrInvalidConfiguration is returned in case the configuration file is
+	// downloaded but has no useful data.
+	ErrInvalidConfiguration = errors.New(`Invalid configuration file.`)
+)
+
+const (
+	// URL of the configuration file. Remember to use HTTPs.
+	remoteConfigURL = `https://s3.amazonaws.com/lantern_config/cloud.1.6.0.yaml.gz`
+)
+
+// pullConfigFile attempts to retrieve a configuration file over the network,
+// then it decompresses it and returns the file's raw bytes.
+func pullConfigFile() ([]byte, error) {
+	var err error
+	var res *http.Response
+
+	// Issuing a post request to download configuration file.
+	if res, err = http.Get(remoteConfigURL); err != nil {
```

Comment:
@oxtoacart (I'm just replying to advance on this issue before I took another task, but this is not a top priority :-))

I just pushed some changes that will enable pulling the cloud configuration file over proxies periodically.

This is currently implemented like this:

* When the library starts, it immediately attempts to pull a fresh configuration directly, if something fails (timed out, access denied by censors, untrusted ssl connection, etc) it will continue with the default configuration.
* When a proxy starts (this will be requested by the Android app), it will fire a goroutine that will periodically check for a new configuration file, this pull will be through the client proxy itself. If any change is detected (based on the `Etag` and `If-None-Match` headers), the client will reload its configuration the same way the original flashlight client does.